control/trajectory_follower_nodes/src/controller_node.cpp

```diff
@@ -142,31 +144,97 @@ bool Controller::isTimeOut()
+  }
+
+  if (!current_odometry_ptr_) {
+    RCLCPP_INFO_THROTTLE(get_logger(), clock, 5000, "Waiting for current odometry.");
+    return {};
+  }
+
+  if (!current_steering_ptr_) {
+    RCLCPP_INFO_THROTTLE(get_logger(), clock, 5000, "Waiting for current steering.");
+    return {};
+  }
+
+  if (!current_accel_ptr_) {
+    RCLCPP_INFO_THROTTLE(get_logger(), clock, 5000, "Waiting for current accel.");
+    return {};
+  }
+
+  trajectory_follower::InputData input_data;
+  input_data.current_trajectory = *current_trajectory_ptr_;
+  input_data.current_odometry = *current_odometry_ptr_;
+  input_data.current_steering = *current_steering_ptr_;
+  input_data.current_accel = *current_accel_ptr_;
+
+  return input_data;
+}
+
+void Controller::initialize(const trajectory_follower::InputData & input_data)
+{
+  // NOTE: How to initialize the controller depends on the lateral and longitudinal controller
+  // algorithm.
+  //       Currently, just run and sync are implemented since the longitudinal controller has to get
+  //       sync_data from the lateral controller before publishing control command.
+  const auto lat_out = lateral_controller_->run(input_data);
+  const auto lon_out = longitudinal_controller_->run(input_data);
+
+  longitudinal_controller_->sync(lat_out.sync_data);
+  lateral_controller_->sync(lon_out.sync_data);
+
+  is_initialized_ = true;
+}
+
 void Controller::callbackTimerControl()
 {
-  // Since the longitudinal uses the convergence information of the steer
-  // with the current trajectory, it is necessary to run the lateral first.
-  // TODO(kosuke55): Do not depend on the order of execution.
-  lateral_controller_->setInputData(input_data_);  // trajectory, odometry, steering
-  const auto lat_out = lateral_controller_->run();
-  lateral_output_ = lat_out ? lat_out : lateral_output_;  // use previous value if none.
-  if (!lateral_output_) return;
-
-  longitudinal_controller_->sync(lateral_output_->sync_data);
-  longitudinal_controller_->setInputData(input_data_);  // trajectory, odometry
-  const auto lon_out = longitudinal_controller_->run();
-  longitudinal_output_ = lon_out ? lon_out : longitudinal_output_;  // use previous value if none.
-  if (!longitudinal_output_) return;
-
-  lateral_controller_->sync(longitudinal_output_->sync_data);
+  // 1. create input data
+  const auto input_data = createInputData(*get_clock());
+  if (!input_data) {
+    RCLCPP_INFO_THROTTLE(
+      get_logger(), *get_clock(), 5000, "Control is skipped since input data is not ready.");
+    return;
+  }
+
+  // 2. check if controllers are ready
+  if (
+    !lateral_controller_->isReady(*input_data) || !longitudinal_controller_->isReady(*input_data)) {
+    RCLCPP_INFO_THROTTLE(
+      get_logger(), *get_clock(), 5000,
+      "Control is skipped since lateral and/or longitudinal controllers are not ready to run.");
+    return;
+  }
+
+  // 3. check if controllers are initialized
+  if (!is_initialized_) {
+    initialize(*input_data);
```

Comment:
I still don't know why the initialize is needed.

Reply:
https://github.com/autowarefoundation/autoware.universe/blob/e5f93be7e391dc8800d91601f2cfdeee4f88caa2/control/trajectory_follower/include/trajectory_follower/sync_data.hpp#L28

As we discussed, by setting is_steer_converged_ false by default, I could remove the initialize function.